src/apps/loan-list/utils/helpers.ts

```diff
@@ -56,24 +65,75 @@ export const queryMatchesFaust = (query: string | null) => {
 
 export const getStackedItems = (
   view: ListView,
-  list: LoanV2[],
+  list: LoanMetaDataType[],
   itemsShown: number,
-  dueDates: string[] | undefined
+  dueDates: string[] | undefined | null[]
 ) => {
-  let returnLoans: LoanV2[] = [];
+  let returnLoans: LoanMetaDataType[] = [];
   if (view === "stacked" && dueDates) {
-    const dueDatesCopy = dueDates.slice(0, itemsShown);
+    // I mean... this...
+    // If the due date is null, the stacked item still has to be shown
+    let dueDatesCopy = [...dueDates, null];
+    dueDatesCopy = dueDatesCopy.slice(0, itemsShown);
     dueDatesCopy.forEach((uniqueDueDate) => {
       returnLoans = returnLoans.concat(
-        list.filter(({ loanDetails }) => loanDetails.dueDate === uniqueDueDate)
+        list.filter(({ dueDate }) => dueDate === uniqueDueDate)
       );
     });
   }
   return returnLoans;
 };
 
-export const getListItems = (list: LoanV2[], itemsShown: number) => {
+export const getListItems = (list: LoanMetaDataType[], itemsShown: number) => {
   return [...list].splice(0, itemsShown);
 };
 
+export const mapPublizonLoanToLoanMetaDataType = (
+  list: Loan[]
+): LoanMetaDataType[] => {
+  return list.map(({ loanExpireDateUtc, orderDateUtc, libraryBook }) => {
+    return {
+      dueDate: loanExpireDateUtc,
+      loanDate: orderDateUtc,
+      id: libraryBook?.identifier || "",
+      isRenewable: false,
+      materialItemNumber: libraryBook?.identifier || "",
+      renewalStatusList: [],
+      loanType: null
+    };
+  });
+};
+
+export const mapPBSLoanToLoanMetaDataType = (
+  list: LoanV2[]
+): LoanMetaDataType[] => {
+  return list.map(({ loanDetails, isRenewable, renewalStatusList }) => {
+    return {
+      dueDate: loanDetails.dueDate,
+      loanDate: loanDetails.loanDate,
+      renewalStatusList,
+      id: loanDetails.recordId,
+      isRenewable,
+      materialItemNumber: loanDetails.materialItemNumber,
+      loanType: loanDetails.loanType
+    };
+  });
+};
+
+export const mapPBSRenewedLoanToLoanMetaDataType = (
+  list: RenewedLoanV2[]
+): LoanMetaDataType[] => {
+  return list.map(({ loanDetails }) => {
```

Comment:
You can also solve this problem by merging objects using destructing. Then you do not have to use a loop.